deltacat/compute/compactor/compaction_session.py

```diff
@@ -56,12 +56,47 @@ def check_preconditions(
     )
 
 
+def check_build_initial_primary_key_indices_preconditions(
```

Comment:
I see this method is retrieving compacted snapshot from a different partition locator. Maybe we can rename accordingly.

Reply:
This method is checking the preconditions before building the primary key indices, like the partition value we build primary key indices on should be the same as the partition value of destination partition we written to. [Here is](https://github.com/ray-project/deltacat/blob/76b49d818e310412347e971a1f470a1d4fc38822/deltacat/compute/compactor/compaction_session.py#L35) the precondition check method for actual compaction.

Reply:
Would you mind elaborate more on what should be the method name?